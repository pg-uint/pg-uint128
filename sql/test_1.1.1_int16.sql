-- Testing cross-types compatibility

SELECT (0::int16)::json;
SELECT json_build_object('some', 0::int16);
SELECT ('0'::json)::int16;
SELECT (0::int16)::jsonb;
SELECT jsonb_build_object('some', 0::int16);
SELECT ('0'::jsonb)::int16;
SELECT ('"abc"'::jsonb)::int16;
