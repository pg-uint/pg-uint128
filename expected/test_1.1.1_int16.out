-- Testing cross-types compatibility
SELECT (0::int16)::json;
 json 
------
 0
(1 row)

SELECT json_build_object('some', 0::int16);
 json_build_object 
-------------------
 {"some" : 0}
(1 row)

SELECT ('0'::json)::int16;
 int16 
-------
 0
(1 row)

SELECT (0::int16)::jsonb;
 jsonb 
-------
 0
(1 row)

SELECT jsonb_build_object('some', 0::int16);
 jsonb_build_object 
--------------------
 {"some": 0}
(1 row)

SELECT ('0'::jsonb)::int16;
 int16 
-------
 0
(1 row)

